src/Uno.UI/UI/Xaml/DragDropManager.cs

```diff
@@ -0,0 +1,136 @@
+		public bool AreConcurrentOperationsEnabled { get; set; } = false;
+
+		/// <inheritdoc />
+		public void BeginDragAndDrop(CoreDragInfo info, ICoreDropOperationTarget? target = null)
+		{
+			if (
+#if __WASM__
+				_window.Dispatcher.IsThreadingSupported &&
+#endif
+				!_window.Dispatcher.HasThreadAccess)
+			{
+				_window.Dispatcher.RunAsync(CoreDispatcherPriority.Normal, () => BeginDragAndDrop(info, target));
+				return;
+			}
+
+			if (!AreConcurrentOperationsEnabled)
+			{
+				foreach (var pending in _dragOperations.ToArray())
+				{
+					pending.Abort();
+				}
+			}
+
+			RegisterWindowHandlers();
+
+			var op = new DragOperation(_window, _hostExtension, info, target);
+
+			_dragOperations.Add(op);
+			info.RegisterCompletedCallback(_ => _dragOperations.Remove(op));
+		}
+
+		/// <summary>
+		/// This method is expected to be invoked each time a pointer involved in a drag operation is moved,
+		/// no matter if the drag operation has been initiated from this app or from an external app.
+		/// </summary>
+		/// <returns>
+		/// The last accepted operation.
+		/// Be aware that due to the async processing of dragging in UWP, this might not be the up to date.
+		/// </returns>
+		public DataPackageOperation ProcessMoved(IDragEventSource src)
+			=> FindOperation(src)?.Moved(src) ?? DataPackageOperation.None;
+
+		/// <summary>
+		/// This method is expected to be invoked when pointer involved in a drag operation is released,
+		/// no matter if the drag operation has been initiated from this app or from an external app.
+		/// </summary>
+		/// <returns>
+		/// The last accepted operation.
+		/// Be aware that due to the async processing of dragging in UWP, this might not be the up to date.
+		/// </returns>
+		public DataPackageOperation ProcessDropped(IDragEventSource src)
+			=> FindOperation(src)?.Dropped(src) ?? DataPackageOperation.None;
+
+		/// <summary>
+		/// This method is expected to be invoked when pointer involved in a drag operation
+		/// is lost for operation initiated by the current app,
+		/// or left the window (a.k.a. the "virtual pointer" is lost) for operation initiated by an other app.
+		/// </summary>
+		/// <returns>
+		/// The last accepted operation.
+		/// Be aware that due to the async processing of dragging in UWP, this might not be the up to date.
+		/// </returns>
+		public DataPackageOperation ProcessAborted(IDragEventSource src)
+			=> FindOperation(src)?.Aborted(src) ?? DataPackageOperation.None;
+
+		private DragOperation? FindOperation(IDragEventSource src)
+			=> _dragOperations.FirstOrDefault(drag =>  drag.Info.SourceId == src.Id);
+
+		private void RegisterWindowHandlers()
+		{
+			if (_areWindowEventsRegistered)
+			{
+				return;
+			}
+
+			// Those events are subscribed for safety, but they are usually useless as:
+			//
+			// # for internally initiated drag operation:
+			//		the pointer is (implicitly) captured by the GestureRecognizer when a Drag manipulation is detected;
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6333952)